packages/bsky/src/views/index.ts

```diff
@@ -385,6 +389,62 @@ export class Views {
     return { count: knownFollowers.count, followers }
   }
 
+  verification(
+    did: string,
+    state: HydrationState,
+  ): VerificationState | undefined {
+    const actor = state.actors?.get(did)
+    if (!actor) return
+
+    const isImpersonation = state.labels?.get(did)?.isImpersonation
+
+    const verifications: VerificationView[] = actor.verifications.map(
+      ({ issuer, uri, displayName, handle, createdAt }) => {
+        // @NOTE: We don't factor-in impersonation when evaluating the validity of each verification,
+        // only in the overall profile verification validity.
+        const isValid =
+          !!displayName &&
+          displayName === actor.profile?.displayName &&
+          !!handle &&
+          handle === actor.handle
+
+        return {
+          issuer,
+          uri,
+          isValid,
+          createdAt,
+        }
+      },
+    )
+    const hasValidVerification = verifications.some((v) => v.isValid)
+
+    const verifiedStatus = hasValidVerification
+      ? isImpersonation
+        ? 'invalid'
+        : 'valid'
+      : 'none'
```

Comment:
I sort of expected the logic here to look something like this:
```suggestion
    const verifiedStatus = verifications.length ?
      ? hasValidVerification && !isImpersonation
        ? 'valid'
        : 'invalid'
      : 'none'
```